src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,47 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+
+type Props = {
+  showModal: boolean;
+  setShowModal: (close: boolean) => void;
+  todo: Todo;
+  setIconEyeId: (iconEyeId: number) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  showModal,
+  setShowModal,
+  todo,
+  setIconEyeId,
+}) => {
+  const [user, setUser] = useState<User>();
+  const [loading, setLoading] = useState(false);
+
+  function getCurrentUser() {
+    getUser(todo.userId)
+      .then(setUser)
+      .finally(() => setLoading(false));
+  }
```

Comment:
Consider adding error handling to the `getCurrentUser` function to manage cases where the user data cannot be fetched. This will improve the robustness of the application.